src/ledger-iframe-bridge.ts

```diff
@@ -112,6 +128,16 @@ export class LedgerIframeBridge implements LedgerBridge {
     }
   }
 
+  getOptions(): LedgerIframeBridgeOptions {
+    return {
+      bridgeUrl: this.bridgeUrl,
+    };
+  }
+
+  setOptions(opts: LedgerIframeBridgeOptions): void {
+    this.bridgeUrl = opts.bridgeUrl ?? BRIDGE_URL;
+  }
```

Comment:
When calling `setOptions` and changing the `bridgeUrl` we should do something to load another iframe, destroying the old one: this looks like destroying the bridge and re-initialising it (so that we don't end up having multiple event listeners that could lead to unpredictable results)

Reply:
i thought setOptions and getOption is mainly to support serialize deserialize, thats why i was using metadata, rather than option

as when you serialize the keyring, u need to deserialize them from the ethKeyringController? 

we only create iframe when init called?

![image](https://github.com/MetaMask/eth-ledger-bridge-keyring/assets/102275989/9b57c26a-9acd-4ced-977a-4282cd64085b)

Reply:
one of the example for using deserialize is in submitEncryptionKey
`submitEncryptionKey -> unlockKeyrings -> #restoreKeyring-> deserialize + init`

It decouple the step for deserialize and init the keyring  , as some high level methods (verifySeedPhrase), they may just wanna to update the data for keyring, but do nothing

i think for iframe recreation, client may just call init after the  to do it after they setOptions

or

we create methods `deserializeData`, `serializeData` in bridge, and leave the `setOptions` and `getOptions` to something that fit to your idea?

```suggestion

  someOtherData = 0;

  async deserializeData(serializeData: Record<string, unknown>) {
      for (const key in serializeData) {
          if (key === 'bridgeUrl') {
            this.bridgeUrl = serializeData[key] as string;
          }
          if (key === 'someOtherData') {
            this.someOtherData = serializeData[key] as number;
          }
      }
  }

  async serializeData(): Promise<Record<string, unknown>> {
    return await this.getOptions()
  }

  async getOptions(): Promise<LedgerIframeBridgeOptions> {
    return Promise.resolve({ bridgeUrl: this.bridgeUrl });
  }

  async setOptions(opts:LedgerIframeBridgeOptions): Promise<void> {
    if (opts.bridgeUrl) {
       this.bridgeUrl = opts.bridgeUrl
       await this.destroy();
       await this.init();
   }
 }
```